TwoHoSun/Screens/ProfileSettings/ProfileSettingViewModel.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  ProfileSettingViewModel.swift
+//  TwoHoSun
+//
+//  Created by 관식 on 10/17/23.
+//
+
+import Alamofire
+import Combine
+import Observation
+import SwiftUI
+
+@Observable
+class ProfileSettingViewModel {
+    var isDuplicated: Bool = false
+    private var cancellable: Set<AnyCancellable> = []
+    
+    func postNickname(nickname: String) {
```

Comment:
```suggestion
       func postNickname(nickname: String) {
        let requestURL = URLConst.baseURL + "/api/profiles/isValidNickname"
        let headers: HTTPHeaders = [
            "Content-Type": "application/json",
            "Authorization": "Bearer \(KeychainManager.shared.readToken(key: "accessToken"))"
        ]
        let body: [String: String] = ["userNickname": nickname]
        
        AF.request(requestURL, method: .post, parameters: body, encoding: JSONEncoding.default, headers: headers)
            .publishDecodable(type: NicknameResponse.self)
            .value()
            .map {$0.data}
            .receive(on: DispatchQueue.main)
            .sink { completion in
                switch completion {
                case .finished:
                    break
                case .failure(let error):
                    print("Error: \(error)")
                }
            } receiveValue: { data in
                self.isDuplicated = data.isExist
                    print("Nickname is duplicated: \(data.isExist)")
            }
            .store(in: &cancellable)
    }
```
이렇게 코드를 수정한다면 switch문을 안놓아도 됩니다. case별로도

Reply:
감사합니다. 반영해서 수정하겠습니다.